tensorflow/contrib/go/graph_test.go

```diff
@@ -0,0 +1,252 @@
+package tensorflow_test
+
+import (
+	"testing"
+
+	tf "github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func TestGraphGeneration(t *testing.T) {
+	graph := tf.NewGraph()
+	input1 := graph.Placeholder("input1", tf.DtInt32, []int64{3}, []string{})
+	input2 := graph.Placeholder("input2", tf.DtInt32, []int64{3}, []string{})
```

Comment:
Reading this I'm a little confused about why the Placeholder allows specifying a per-dimension name.  It doesn't really match the tf.placeholder python version, and none of the example or test code actually exercises this functionality.  Can it just be removed, or does it serve a purpose that's not illustrated in here?  If the latter, it would be good to add a test/example that shows it.